src/crypto/keys.rs

```diff
@@ -174,6 +174,46 @@ pub struct PublicKey {
 
 
 impl PublicKey {
+    /// Creates an RSA Public Key based on the supplied exponent and modulus.
+    /// 
+    /// See:
+    /// [RFC 4055]: https://tools.ietf.org/html/rfc4055
+    /// 
+    /// An RSA Public Key uses the following DER encoded structure inside its
+    /// BitString component:
+    /// 
+    /// ```txt
+    /// RSAPublicKey  ::=  SEQUENCE  {
+    ///     modulus            INTEGER,    -- n
+    ///     publicExponent     INTEGER  }  -- e
+    /// ```
+    pub fn rsa_from_components(
+        modulus: &[u8], // n
+        exponent: &[u8] // e 
+    ) -> Result<Self, bcder::decode::Error> {
+        let modulus = bcder::Unsigned::from_slice(modulus)?;
+        let exponent = bcder::Unsigned::from_slice(exponent)?;
+
+        let pub_key_sequence = bcder::encode::sequence((
+            modulus.encode(),
+            exponent.encode()
+        ));
+
+        Ok(PublicKey::rsa_from_public_key_bytes(
+            pub_key_sequence.to_captured(bcder::Mode::Der).into_bytes()
+        ))
+    }
+
+    /// Creates an RSA Public Key based on the given RSAPublicKey bytes.
+    pub fn rsa_from_public_key_bytes(
```

Comment:
Should this check that the bytes are correctly encoded?

Reply:
Perhaps.. both functions are taken from the Krill code where it deals with various flavours of public key bytes being returned using KMIP. That code did not very that those bytes make actual sense - though things would fail somewhere if they don't.

We could check here that it's at least a sequence of modulus and exponent. But of course it would be better still to check that those values also make sense as such. I am not sure how to do the latter - but I suppose I could find out.

Reply:
You don’t need the check for `rsa_from_components` as that will always make a proper object. But the function is pub, so it should probably check since `PublicKey` should always be a valid public key.